### PR TITLE
Putnamos aka "i made LINDA fast"

### PR DESCRIPTION
**Pull request by Putnam3145**
## About The Pull Request

Made LINDA work by just mixing a tile with its neighbors instantly and fully instead of all this wacky "share" bull. Should make things way, way faster, in many ways.

**TESTMERGE THIS OH MY GOD**

## Why It's Good For The Game

LINDA is super bad. It takes this baffling perfect approach to simulation when good-enough is... good enough, and much faster. This is good-enough.

## Changelog
:cl:
tweak: Putnamos: faster faster faster.
/:cl:

### CODE REVIEW

File: code/__DEFINES/atmospherics.dm [hunk 1]
```diff
@@ -149,9 +149,9 @@
 
 //OPEN TURF ATMOS
 #define OPENTURF_DEFAULT_ATMOS		"o2=22;n2=82;TEMP=293.15" //the default air mix that open turfs spawn
-#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
+#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15�C telecommunications. also used for xenobiology slime killrooms
```
Comment (Putnam3145) [inline code/__DEFINES/atmospherics.dm]: god DAMMIT

**Comment by deathride58** [comment]
WORK IT MAKE IT DO IT MAKES US HARDER BETTER FASTER STRONGER

**Comment by Arturlang** [comment]
I am very afraid AND excited

File: code/__DEFINES/atmospherics.dm (hunk 1 above)
Comment (Ghommie) [inline code/__DEFINES/atmospherics.dm]:
```suggestion
#define TCOMMS_ATMOS				"n2=100;TEMP=80" //-193,15°C telecommunications. also used for xenobiology slime killrooms
```
Deja vu.

File: code/__DEFINES/atmospherics.dm [hunk 2]
```diff
@@ -149,9 +149,9 @@
...
 #define AIRLESS_ATMOS				"TEMP=2.7" //space
-#define FROZEN_ATMOS				"o2=22;n2=82;TEMP=180" //-93.15�C snow and ice turfs
+#define FROZEN_ATMOS				"o2=22;n2=82;TEMP=180" //-93.15�C snow and ice turfs
```
Comment (Ghommie) [inline code/__DEFINES/atmospherics.dm]:
```suggestion
#define FROZEN_ATMOS				"o2=22;n2=82;TEMP=180" //-93.15°C snow and ice turfs
```

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 3]
```diff
@@ -199,30 +209,43 @@
...
+		var/datum/gas_mixture/final_air = new()
+		final_air.merge(our_air.copy())
+		for(var/t in adjacent_turfs)
+			var/turf/open/enemy_tile = t
+			final_air.merge(enemy_tile.air.copy())
+		if (planet_atmos) //share our air with the "atmosphere" "above" the turf
+			var/datum/gas_mixture/G = new
+			G.copy_from_turf(src)
+			ARCHIVE_TEMPERATURE(G)
+			if(!(our_air.compare(G)))
```
Comment (Ghommie) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm]: Are the parethesis after the ! operator really needed?

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm [hunk 4]
```diff
@@ -154,28 +157,35 @@
...
 	var/adjacent_turfs_length = LAZYLEN(adjacent_turfs)
 	var/cached_atmos_cooldown = atmos_cooldown + 1
 
 	var/planet_atmos = planetary_atmos
+	var/datum/gas_mixture/our_air = air
 	if (planet_atmos)
+		var/datum/gas_mixture/G = new
+		G.copy_from_turf(src)
+		ARCHIVE_TEMPERATURE(G)
+		if(!(our_air.compare(G)))
```
Comment (Ghommie) [inline code/modules/atmospherics/environmental/LINDA_turf_tile.dm]: Same.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Minor things, not a ss13 atmos code expert.

**Comment by Poojawa** [comment]
have you trialed this with holodeck burn sim?

have you stress tested with dozens of hull breaches and canister openings, etc etc. 

Def put up a recording of some of what happens in local.

File: code/controllers/subsystem/air.dm [hunk 5]
```diff
@@ -44,6 +45,12 @@ SUBSYSTEM_DEF(air)
 	var/map_loading = TRUE
 	var/list/queued_for_activation
 
+	// VORELINDA config options (am I really calling it that?)
+	// The below two refer to how many SSair_turf ticks will go by before a turf will run again.
```
Comment (Poojawa) [inline code/controllers/subsystem/air.dm:49]: I'm getting prob(80) vibes

File: code/modules/atmospherics/environmental/LINDA_turf_tile.dm (hunk 3 above)
Comment (Poojawa) [reply code/modules/atmospherics/environmental/LINDA_turf_tile.dm]: they don't hurt for sure.

File: code/controllers/subsystem/air.dm [hunk 6]
```diff
@@ -44,6 +45,12 @@ SUBSYSTEM_DEF(air)
 	var/map_loading = TRUE
 	var/list/queued_for_activation
 
+	// VORELINDA config options (am I really calling it that?)
```
Comment (Ghommie) [inline code/controllers/subsystem/air.dm]: > VORELINDA

Any other name that doesn't explicitly reference that weird fetish, like LINDA+, V-LINDA, PUT-LINDA, P-LINDA, CIT-LINDA, would be appreciated.

File: code/controllers/subsystem/air.dm (hunk 5 above)
Comment (Ghommie) [reply code/controllers/subsystem/air.dm:49]: Thing is `prob(80)` has no fixed, consistent (per tick) frequency, unlike here.

Comment (deathride58) [reply code/controllers/subsystem/air.dm:49]: This is there primarily to artificially slow down the simulation. Without it comes quite a few gameplay issues that come from TG's map design... *really* not being meant for ZAS propagation speeds

File: code/__DEFINES/atmospherics.dm (hunk 1 above)
Comment (Putnam3145) [reply code/__DEFINES/atmospherics.dm]: can we just replace it with the word "degrees"

**Review by Ghommie — APPROVED** [review]

File: code/controllers/subsystem/air.dm (hunk 6 above)
Comment (lolman360) [reply code/controllers/subsystem/air.dm]: LINDA-next

**Comment by Ghommie** [comment]
Will this ever be ready?

**Comment by Putnam3145** [comment]
Not likely, I've kinda given up on it

**Comment by Ghommie** [comment]
So, should I close it?

**Pull request closed without merging**